### ISSUE TITLE
[BUG] User counters only showing results for first five of total set of permutations

### ISSUE THREAD

**Issue by PhilipDeegan** [google/benchmark#1306]
**Describe the bug**
I'm trying to use libpfm and the performance counters offered by this library, but it only appears to give me some of the results

**System**
Which OS, compiler, and compiler version are you using:
  - OS: Debian 11
  - Compiler and version:  gcc 10.2

**To reproduce**
1. sync to commit master
2. compile example with "-DHAVE_LIBPFM" / link libpfm
3. run with "--benchmark_perf_counters=CYCLES"
4. See output

**Expected behavior**
Metrics are given for all permutations

**Screenshots**
![image](https://user-images.githubusercontent.com/30491/145996885-60805b68-fee7-4b12-8793-03eec4c99071.png)
![image](https://user-images.githubusercontent.com/30491/145996902-b5899b4a-b367-45cb-b77f-0bdc2d6cc27d.png)

**Additional context**
sample reproducible 

```
#include <cstdlib>
#include "benchmark/benchmark.h"

static std::vector<std::int64_t> A{1, 10};
static std::vector<std::int64_t> B{100, 500, 1000};
static std::vector<std::int64_t> C{50, 100, 150, 200};

template<std::size_t X, std::size_t Y, bool B>
struct Fixture : public benchmark::Fixture
{
public:
    void SetUp(::benchmark::State const& state) override
    {
        std::uint32_t a = state.range(0);
        std::uint32_t b = state.range(1);

        doubles.clear();
        for (std::size_t i = 0; i < a; ++i)
            doubles.emplace_back(b);
    }

    void run(::benchmark::State& state)
    {
        for (auto _ : state)
            for (auto& vec : doubles)
                for (auto& d : vec)
                    d += 1;
    }

    std::vector<std::vector<double>> doubles;
};

BENCHMARK_TEMPLATE_DEFINE_F(Fixture, _3_1_1, 3, 1, 1)(benchmark::State& state)
{
    run(state);
}
BENCHMARK_REGISTER_F(Fixture, _3_1_1)->Unit(benchmark::kNanosecond)->ArgsProduct({A, B, C});

BENCHMARK_TEMPLATE_DEFINE_F(Fixture, _3_1_0, 3, 1, 0)(benchmark::State& state)
{
    run(state);
}
BENCHMARK_REGISTER_F(Fixture, _3_1_0)->Unit(benchmark::kNanosecond)->ArgsProduct({A, B, C});

int main(int argc, char** argv)
{
    ::benchmark::Initialize(&argc, argv);
    ::benchmark::RunSpecifiedBenchmarks();
}
```

**Comment by dominichamon** [comment]
@mtrofin ^^

**Comment by mtrofin** [comment]
@taoliq, who identified this issue and was making progress on fixing it.

He bisected it to fbc3140, and has a fix (https://github.com/taoliq/benchmark/commit/3658cbe1b9a2982603526ac7b6f6ab38e90ec8f9) but we were trying to nail down the root cause.

@PhilipDeegan, to unblock, could you try @taoliq's fix?

**Comment by PhilipDeegan** [comment]
> @taoliq, who identified this issue and was making progress on fixing it.
> 
> He bisected it to [fbc3140](https://github.com/google/benchmark/commit/fbc31405b2a2838560e155d7aec937e135ae0d81), and has a fix ([taoliq@3658cbe](https://github.com/taoliq/benchmark/commit/3658cbe1b9a2982603526ac7b6f6ab38e90ec8f9)) but we were trying to nail down the root cause.
> 
> @PhilipDeegan, to unblock, could you try @taoliq's fix?

it appears to work, there's a type here tho 

https://github.com/taoliq/benchmark/blob/3658cbe1b9a2982603526ac7b6f6ab38e90ec8f9/src/benchmark_runner.cc#L169

it should be ```perf_counters_measurement_ptr->IsValid())``` I suppose
